### ISSUE TITLE
RUF015 bad code example does not throw a warning

### ISSUE THREAD

**Issue by jschippergoauto** [astral-sh/ruff#5764]
Ruff version: 0.0.278
Command: `ruff --config=ruff.toml file.py`
`ruff.toml`:
```toml
extend-select = [
  "RUF",
]
```
Code snippet:
```python
head = list(range(1000000000000))[0]
```

The code snippet was taken from the [Ruff documentation](https://beta.ruff.rs/docs/rules/unnecessary-iterable-allocation-for-first-element/)

I expected it to give me a RUF015 warning but instead it completely ignores it. Changing the snippet to the following does throw a warning.
```python
x = range(1000000000000)
head = list(x)[0]
```
